setup.cfg

```diff
@@ -81,6 +81,51 @@ setup_requires =
     setuptools >= 38.4
     setuptools_scm
 
+[options.extras_require]
+io =
+    netCDF4
+    h5netcdf
+    scipy
+    pydap
+    zarr
+    fsspec
+    cftime
+    rasterio
+    cfgrib
+    ## Scitools packages & dependencies (for example: cartopy, cf-units) can be hard to install
+    # scitools-iris
+    ## Required by scitools-iris
+    # scitools-pyke
+
+accel =
+    scipy
+    bottleneck
+    numbagg
+
+parallel =
+    dask[array,dataframe]
+
+viz =
+    matplotlib
+    seaborn
+    nc-time-axis
+    ## Cartopy & dependencies require 3rd party libraries and only provides source distributions
+    ## See: https://github.com/SciTools/cartopy/issues/805
+    # cartopy
+
+complete =
+    %(io)s
+    %(accel)s
+    %(parallel)s
+    %(viz)s
+
+docs =
+    %(complete)s
+    sphinx-autosummary-accessors
+    sphinx_rtd_theme
+    ipython
+    nbsphinx
```

Comment:
[`doc.yml`](https://github.com/pydata/xarray/blob/45aab423b61c99278dd9b4d8a21915647afae623/ci/requirements/doc.yml) also lists `ipykernel`, `jupyter_client` and `scanpydoc` (the last one is new on `master`). Could we add those, too?

Reply:
Will do. Should we call it `[docs]` or `[doc]` - which do you prefer?

Reply:
what I meant was that `ipykernel`, `jupyter_client` and `scanpydoc` should be added to the existing docs section.

I think `doc.yml` is named that way because the directory containing the docs is called `doc`. That does not mean that we have to copy so keeping the current name should be fine, too.